src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocPackageCheck.java

```diff
@@ -69,7 +69,7 @@ public void beginProcessing(String charset) {
     @Override
     protected void processFiltered(File file, FileText fileText) {
         // Check if already processed directory
-        final File dir = file.getParentFile();
+        final File dir = fileText.getFile().getParentFile();
```

Comment:
"protected void processFiltered(File file, FileText fileText)" is a signature that all Checks have.
So it is still possible to call `file.getParentFile();` and got NPE. 
So this is not a fix for a problem in general, it is just custom fix for custom Check.
If that is not a fix , why we change api ?
It is a problem of certain collection + nuance with `file.getParentFile`. Check need to handle it himself as it did before 8.2.

Reply:
In fact, first argument in "protected void processFiltered(File file, FileText fileText)" is redundant, see #5196. My idea was to insure that if FileText could have been created it contains full info about file path. And then all checks (assuming #5196 would be approved) would have correct file info by definition.

If such assumptions are premature I will just fix this particular Check.

Reply:
Please fix Check in scope of this issue.
#5196 should be done as we finish elimination of all Holders and complete merge if AST filter

Reply:
Done